.github/workflows/dev-setup.yml

```diff
@@ -0,0 +1,45 @@
+name: Dev environment setup
+
+on:
+  schedule:
+    - cron: "42 7 * * 0"
+  workflow_dispatch:
+    inputs:
+      repository:
+        description: >
+          [Optional] Repository name with owner. For example, mlflow/mlflow.
+           Defaults to the repository that triggered a workflow.
+        required: false
+        default: ""
+      ref:
+        description: >
+          [Optional] The branch, tag or SHA to checkout. When checking out the repository that
+           triggered a workflow, this defaults to the reference or SHA for that event. Otherwise,
+           uses the default branch.
+        required: false
+        default: ""
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref == 'refs/heads/master' && github.run_number || github.ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash --noprofile --norc -exo pipefail {0}
+
+jobs:
+  linux-env-setup:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          repository: ${{ github.event.inputs.repository }}
+          ref: ${{ github.event.inputs.ref }}
+      - uses: ./.github/actions/setup-pyenv
+      - name: Setup environment
+        run: |
+          git config --global user.name "test"
+          git config --global user.email "test@mlflow.org"
+      - name: Run Environment tests
+        run: |
+          TERM=xterm bash ./dev/test-dev-env-setup.sh
```

Comment:
Is `TERM=xterm` only required on github?

Reply:
yes. In a local terminal client there will be some defined terminal value to allow for syntactic formatting. I took this tip directly from a Github dev on an issue that they responded on that recommended specifying this if using any sort of tput commands within the script.